generic/tests/unattended_install.py

```diff
@@ -11,4 +18,52 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters.
     :param env: Dictionary with test environment.
     """
-    unattended_install.run(test, params, env)
+    # use vm name to differentiate among single/multivm
+    vms = params.get("vms").split()
+
+    if len(vms) < 2:
+        unattended_install.run(test, params, env)
+        return
+
+    def thread_func(vm):
+        """
+        Thread Method to trigger the unattended installation
+
+        :param vm: VM name
+        """
+        global error_flag
+        try:
+            vm_params = params.object_params(vm)
+            vm_params["main_vm"] = vm
+            unattended_install.run(test, vm_params, env)
+        except Exception as info:
+            logging.error(info)
+            error_flag = True
+
+    trigger_time = int(params.get("install_trigger_time", 0))
+    random_trigger = params.get("random_trigger", "no") == "yes"
+    install_threads = []
+
+    if not params.get("master_images_clone"):
+        test.cancel("provide the param `master_images_clone` to clone"
+                    "images for vms")
+
+    for vm in vms:
+        thread = threading.Thread(target=thread_func, args=(vm,))
+        install_threads.append(thread)
+
+    for thread in install_threads:
+        if trigger_time:
+            sleep_time = trigger_time
+            if random_trigger:
+                sleep_time = random.randint(0, trigger_time)
+            time.sleep(sleep_time)
+        thread.start()
+
+    for thread in install_threads:
+        thread.join()
+        if thread.isAlive():
+            test.error("Unattended install %s timeout" % thread)
```

Comment:
Then we don't need this check anymore since it won't happen.

Reply:
made the changes :+1: